tests/test/kube.rs

```diff
@@ -362,11 +363,24 @@ where
 }
 
 /// Deserializes the given YAML text into the desired type.
-pub fn from_yaml<T>(str: &str) -> T
+pub fn from_yaml<T>(yaml: &str) -> T
 where
     T: DeserializeOwned,
 {
-    serde_yaml::from_str(str).expect("String is not a well-formed YAML")
+    serde_yaml::from_str(yaml).expect("String is not a well-formed YAML")
+}
+
+/// Appends a UUID to `metadata/name`.
+pub fn with_unique_name(yaml: &str) -> String {
```

Comment:
You know about the generateName functionality from Kubernetes?

Reply:
I think that generateName is not appropriate for the test case [restart_after_ungraceful_shutdown_should_succeed](https://github.com/stackabletech/agent-integration-tests/blob/4e9a210dc3e4fc81d26143381b662ac854b3c016/tests/service.rs#L39-L60).
I want a unique name in case several tests run in parallel but the pod should be "restarted" with the same name.

Reply:
Makes sense, thank you!